docs/decisions/0024-planner-telemetry-enhancement.md

```diff
@@ -0,0 +1,366 @@
+
+- The framework should be telemetry service agnostic.
+- Enabling and disabling specific telemetry items should require minimum effort.
+
+- The following metrics should be emitted by SK:
+  - Input token usage for prompt
+    - Description: A prompt is the smallest unit that consumes tokens. A function can consist of multiple prompts.
+    - Dimensions: ComponentType, ComponentName, Service ID, Model ID
+    - Example:
+      | ComponentType | ComponentName | Service ID | Model ID | Value |
+      |---|---|---|---|---|
+      | Function | WritePoem | | GPT-3.5-Turbo | 40
+      | Function | TellJoke | | GPT-4 | 50
+      | Function | WriteAndTellJoke | | GPT-3.5-Turbo | 30
+      | Planner | CreateHandlebarsPlan | | GPT-3.5-Turbo | 100
+  - Output token usage for prompt
+    - Description: A prompt is the smallest unit that consumes tokens. A function can consist of multiple prompts.
+    - Dimensions: ComponentType, ComponentName, Service ID, Model ID
+    - Example:
+      | ComponentType | ComponentName | Service ID | Model ID | Value |
+      |---|---|---|---|---|
+      | Function | WritePoem | | GPT-3.5-Turbo | 40
+      | Function | TellJoke | | GPT-4 | 50
+      | Function | WriteAndTellJoke | | GPT-3.5-Turbo | 30
+      | Planner | CreateHandlebarsPlan | | GPT-3.5-Turbo | 100
+  - Model request time for prompt
+    - Description: A prompt is the smallest unit that consumes tokens. A function can consist of multiple prompts.
+    - Dimensions: ComponentType, ComponentName, Service ID, Model ID
+    - Example:
+      | ComponentType | ComponentName | Service ID | Model ID | Value |
+      |---|---|---|---|---|
+      | Function | WritePoem | | GPT-3.5-Turbo | 0.5
+      | Function | TellJoke | | GPT-4 | 0.5m
+      | Function | WriteAndTellJoke | | GPT-3.5-Turbo | 0.3m
+      | Planner | CreateHandlebarsPlan | | GPT-3.5-Turbo | 1m
+  - Aggregated execution time for functions
+    - Description: A function can consist of multiple prompts. The execution time of a function is the duration from start to end of a function's `invoke` call.
+    - Dimensions: ComponentType, ComponentName, Service ID, Model ID
+    - Example:
+      | ComponentType | ComponentName | Value |
+      |---|---|---|
+      | Function | WritePoem | 1m
+      | Function | TellJoke | 1m
+      | Function | WriteAndTellJoke | 1.5m
+      | Planner | CreateHandlebarsPlan | 2m
+  - Success count for planners
+    - Description: A planner run is considered successful when it generates a valid plan.
+    - Dimensions: ComponentType, ComponentName, Service ID, Model ID
+    - Example:
+      | ComponentType | ComponentName | Fail | Success
+      |---|---|---|---|
+      | Planner | CreateHandlebarsPlan | 5 | 95
+      | Planner | CreateHSequentialPlan | 20 | 80
+  - Success count for plans
+    - Description: A plan execution is considered successful when all steps in the plan are executed successfully.
+    - Dimensions: ComponentType, ComponentName, Service ID, Model ID
```

Comment:
As above we need to know which plan failed. 

It would also be useful to know which step in the plan failed.

Reply:
We need to go into the trace detail to figure out the failure details. Metrics are not good medium to do that.